users/views.py

```diff
@@ -0,0 +1,82 @@
+from rest_framework.views import APIView
+import rest_framework.permissions as p
+from .serializers import UserSerializer
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.decorators import api_view, permission_classes
+from . models import User
+from rest_framework_simplejwt.tokens import RefreshToken
+from django.contrib.auth.signals import user_logged_in
+from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+
+def get_tokens_for_user(user):
+    refresh = RefreshToken.for_user(user)
+
+    return {
+        'refresh': str(refresh),
+        'access': str(refresh.access_token),
+    }
+
+
+class CreateUserAPIView(APIView):
+    # Allow any user (authenticated or not) to access this url
+    permission_classes = (p.AllowAny,)
+
+    def post(self, request):
+        user = request.data
+        serializer = UserSerializer(data=user)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+@api_view(['POST'])
+@permission_classes([p.AllowAny, ])
+def authenticate_user(request):
+    try:
+        email = request.data['email']
+        password = request.data['password']
+        user = User.objects.get(email=email, password=password)
+        if user:
+            try:
+                tokens = get_tokens_for_user(user)
+                user_details = {'name': "%s %s" % (
+                    user.first_name, user.last_name), 'token': tokens["access"], 'refresh_token': tokens["refresh"]}
+                #TODO figure out purpose of below
+                user_logged_in.send(sender=user.__class__,
+                                    request=request, user=user)
+                return Response(user_details, status=status.HTTP_200_OK)
+            except Exception as e:
+                raise e
```

Comment:
here you need to return an error with the status 500

Reply:
500 error would be returned anyway in case of exception, wouldn't it?
On the other hand, try-except clause is not necessary unless logging or other exception handling, so I would rather get rid of it for now.

Reply:
Ok, remove try-except

Reply:
Done.